The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/13276429823/job/37066781875) [(plain)](https://github.com/rust-lang/rust/commit/59f491bff19f3337f53b1242a88704babee7c79d/checks/37066781875/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#22 exporting to docker image format
#22 sending tarball 27.4s done
#22 DONE 33.1s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
failures:

---- [codegen] tests/codegen/enum/enum-debug-niche-2.rs stdout ----

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-18/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/enum/enum-debug-niche-2/enum-debug-niche-2.ll" "/checkout/tests/codegen/enum/enum-debug-niche-2.rs" "--check-prefix=CHECK" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
/checkout/tests/codegen/enum/enum-debug-niche-2.rs:8:11: error: CHECK: expected string not found in input
/checkout/tests/codegen/enum/enum-debug-niche-2.rs:8:11: error: CHECK: expected string not found in input
// CHECK: {{.*}}DIDerivedType{{.*}}tag: DW_TAG_member,{{.*}}name: "Placeholder",{{.*}}extraData: i128 4294967295{{[,)].*}}
          ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/enum/enum-debug-niche-2/enum-debug-niche-2.ll:385:200: note: scanning from here
!178 = !DICompositeType(tag: DW_TAG_variant_part, scope: !176, file: !2, size: 32, align: 32, elements: !179, templateParams: !23, identifier: "374a1196b7c1cc46b226c9bff6b392d6", discriminator: !209)
                                                                                                                                                                                                       ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/enum/enum-debug-niche-2/enum-debug-niche-2.ll:409:6: note: possible intended match here
!202 = !DIDerivedType(tag: DW_TAG_member, name: "Placeholder", scope: !178, file: !2, baseType: !203, size: 32, align: 32, extraData: i32 -1)

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/enum/enum-debug-niche-2/enum-debug-niche-2.ll
Check file: /checkout/tests/codegen/enum/enum-debug-niche-2.rs


-dump-input=help explains the following input dump.
Input was:
<<<<<<
           .
           .
           .
           .
         285: !78 = !DISubroutineType(types: !79) 
         286: !79 = !{!80, !81} 
         287: !80 = !DIBasicType(name: "i32", size: 32, encoding: DW_ATE_signed) 
         288: !81 = !DIDerivedType(tag: DW_TAG_pointer_type, name: "&std::rt::lang_start::{closure_env#0}<()>", baseType: !14, size: 64, align: 64, dwarfAddressSpace: 0) 
         289: !82 = !{!83} 
         290: !83 = !DILocalVariable(name: "main", scope: !77, file: !56, line: 193, type: !20, align: 64) 
         291: !84 = !DILocation(line: 193, column: 5, scope: !77) 
         292: !85 = !DILocation(line: 199, column: 70, scope: !77) 
         293: !86 = !DILocation(line: 199, column: 18, scope: !77) 
         294: !87 = !DILocalVariable(name: "self", arg: 1, scope: !88, file: !34, line: 2060, type: !39) 
         295: !88 = distinct !DISubprogram(name: "to_i32", linkageName: "_RNvMsN_NtCs6FUsJgVcnjc_3std7processNtB5_8ExitCode6to_i32", scope: !39, file: !34, line: 2060, type: !89, scopeLine: 2060, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !30, templateParams: !23, declaration: !91, retainedNodes: !92) 
         296: !89 = !DISubroutineType(types: !90) 
         297: !90 = !{!80, !39} 
         298: !91 = !DISubprogram(name: "to_i32", linkageName: "_RNvMsN_NtCs6FUsJgVcnjc_3std7processNtB5_8ExitCode6to_i32", scope: !39, file: !34, line: 2060, type: !89, scopeLine: 2060, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagOptimized, templateParams: !23) 
         299: !92 = !{!87} 
         300: !93 = !DILocation(line: 2060, column: 19, scope: !88, inlinedAt: !94) 
         301: !94 = !DILocation(line: 199, column: 85, scope: !77) 
         302: !95 = !DILocation(line: 636, column: 9, scope: !96, inlinedAt: !102) 
         303: !96 = distinct !DISubprogram(name: "as_i32", linkageName: "_RNvMsb_NtNtNtNtNtCs6FUsJgVcnjc_3std3sys3pal4unix7process14process_commonNtB5_8ExitCode6as_i32", scope: !42, file: !97, line: 635, type: !98, scopeLine: 635, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !30, templateParams: !23, declaration: !101) 
         304: !97 = !DIFile(filename: "/rustc/FAKE_PREFIX/library/std/src/sys/pal/unix/process/process_common.rs", directory: "", checksumkind: CSK_MD5, checksum: "7107dec5baaefd58adc486b058fd5a71") 
         305: !98 = !DISubroutineType(types: !99) 
         306: !99 = !{!80, !100} 
         307: !100 = !DIDerivedType(tag: DW_TAG_pointer_type, name: "&std::sys::pal::unix::process::process_common::ExitCode", baseType: !42, size: 64, align: 64, dwarfAddressSpace: 0) 
         308: !101 = !DISubprogram(name: "as_i32", linkageName: "_RNvMsb_NtNtNtNtNtCs6FUsJgVcnjc_3std3sys3pal4unix7process14process_commonNtB5_8ExitCode6as_i32", scope: !42, file: !97, line: 635, type: !98, scopeLine: 635, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagOptimized, templateParams: !23) 
         309: !102 = !DILocation(line: 2061, column: 16, scope: !88, inlinedAt: !94) 
         310: !103 = !DILocation(line: 199, column: 93, scope: !77) 
         311: !104 = distinct !DISubprogram(name: "__rust_begin_short_backtrace<fn(), ()>", linkageName: "_ZN3std3sys9backtrace28__rust_begin_short_backtrace17heb6544cb70052419E", scope: !106, file: !105, line: 148, type: !107, scopeLine: 148, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !30, templateParams: !113, retainedNodes: !109) 
         312: !105 = !DIFile(filename: "/rustc/FAKE_PREFIX/library/std/src/sys/backtrace.rs", directory: "", checksumkind: CSK_MD5, checksum: "9e30c70624c3cf40238860e740bd696f") 
         313: !106 = !DINamespace(name: "backtrace", scope: !47) 
         314: !107 = !DISubroutineType(types: !108) 
         315: !108 = !{null, !20} 
         316: !109 = !{!110, !111} 
         317: !110 = !DILocalVariable(name: "f", arg: 1, scope: !104, file: !105, line: 148, type: !20) 
         318: !111 = !DILocalVariable(name: "result", scope: !112, file: !105, line: 152, type: !7, align: 8) 
         319: !112 = distinct !DILexicalBlock(scope: !104, file: !105, line: 152, column: 5) 
         320: !113 = !{!114, !68} 
         321: !114 = !DITemplateTypeParameter(name: "F", type: !20) 
         322: !115 = !DILocation(line: 152, column: 9, scope: !112) 
         323: !116 = !DILocation(line: 148, column: 43, scope: !104) 
         324: !117 = !DILocalVariable(name: "dummy", scope: !118, file: !119, line: 476, type: !7, align: 8) 
         325: !118 = distinct !DISubprogram(name: "black_box<()>", linkageName: "_ZN4core4hint9black_box17hf080e862270e1f17E", scope: !120, file: !119, line: 476, type: !122, scopeLine: 476, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !30, templateParams: !67, retainedNodes: !124) 
         326: !119 = !DIFile(filename: "/rustc/FAKE_PREFIX/library/core/src/hint.rs", directory: "", checksumkind: CSK_MD5, checksum: "38cd136a2222298e5e1abe826af11780") 
         327: !120 = !DINamespace(name: "hint", scope: !121) 
         328: !121 = !DINamespace(name: "core", scope: null) 
         329: !122 = !DISubroutineType(types: !123) 
         330: !123 = !{null, !7} 
         331: !124 = !{!117} 
         332: !125 = !DILocation(line: 476, column: 27, scope: !118, inlinedAt: !126) 
         333: !126 = !DILocation(line: 155, column: 5, scope: !112) 
         334: !127 = !DILocation(line: 152, column: 18, scope: !104) 
         335: !128 = !DILocation(line: 477, column: 5, scope: !118, inlinedAt: !126) 
         336: !129 = !{i64 7398798428621064} 
         337: !130 = !DILocation(line: 158, column: 2, scope: !104) 
         338: !131 = distinct !DISubprogram(name: "call_once<std::rt::lang_start::{closure_env#0}<()>, ()>", linkageName: "_ZN4core3ops8function6FnOnce40call_once$u7b$$u7b$vtable.shim$u7d$$u7d$17h7639774970898cadE", scope: !133, file: !132, line: 250, type: !136, scopeLine: 250, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !30, templateParams: !142, retainedNodes: !139) 
         339: !132 = !DIFile(filename: "/rustc/FAKE_PREFIX/library/core/src/ops/function.rs", directory: "", checksumkind: CSK_MD5, checksum: "27f40bbdeb6cc525c0d0d7cf434d92c4") 
         340: !133 = !DINamespace(name: "FnOnce", scope: !134) 
         341: !134 = !DINamespace(name: "function", scope: !135) 
         342: !135 = !DINamespace(name: "ops", scope: !121) 
         343: !136 = !DISubroutineType(types: !137) 
         344: !137 = !{!80, !138} 
         345: !138 = !DIDerivedType(tag: DW_TAG_pointer_type, name: "*mut std::rt::lang_start::{closure_env#0}<()>", baseType: !14, size: 64, align: 64, dwarfAddressSpace: 0) 
         346: !139 = !{!140, !141} 
         347: !140 = !DILocalVariable(arg: 1, scope: !131, file: !132, line: 250, type: !138) 
         348: !141 = !DILocalVariable(arg: 2, scope: !131, file: !132, line: 250, type: !7) 
         349: !142 = !{!143, !144} 
         350: !143 = !DITemplateTypeParameter(name: "Self", type: !14) 
         351: !144 = !DITemplateTypeParameter(name: "Args", type: !7) 
         352: !145 = !DILocation(line: 250, column: 5, scope: !131) 
         353: !146 = distinct !DISubprogram(name: "call_once<fn(), ()>", linkageName: "_ZN4core3ops8function6FnOnce9call_once17h90b83f233786f257E", scope: !133, file: !132, line: 250, type: !107, scopeLine: 250, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !30, templateParams: !150, retainedNodes: !147) 
         354: !147 = !{!148, !149} 
         355: !148 = !DILocalVariable(arg: 1, scope: !146, file: !132, line: 250, type: !20) 
         356: !149 = !DILocalVariable(arg: 2, scope: !146, file: !132, line: 250, type: !7) 
         357: !150 = !{!151, !144} 
         358: !151 = !DITemplateTypeParameter(name: "Self", type: !20) 
         359: !152 = !DILocation(line: 250, column: 5, scope: !146) 
         360: !153 = distinct !DISubprogram(name: "call_once<std::rt::lang_start::{closure_env#0}<()>, ()>", linkageName: "_ZN4core3ops8function6FnOnce9call_once17he0925881672264cfE", scope: !133, file: !132, line: 250, type: !154, scopeLine: 250, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !30, templateParams: !142, retainedNodes: !156) 
         361: !154 = !DISubroutineType(types: !155) 
         362: !155 = !{!80, !14} 
         363: !156 = !{!157, !158} 
         364: !157 = !DILocalVariable(arg: 1, scope: !153, file: !132, line: 250, type: !14) 
         365: !158 = !DILocalVariable(arg: 2, scope: !153, file: !132, line: 250, type: !7) 
         366: !159 = !DILocation(line: 250, column: 5, scope: !153) 
         367: !160 = distinct !DISubprogram(name: "drop_in_place<std::rt::lang_start::{closure_env#0}<()>>", linkageName: "_ZN4core3ptr85drop_in_place$LT$std..rt..lang_start$LT$$LP$$RP$$GT$..$u7b$$u7b$closure$u7d$$u7d$$GT$17hee24f019903df49cE", scope: !162, file: !161, line: 523, type: !163, scopeLine: 523, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !30, templateParams: !167, retainedNodes: !165) 
         368: !161 = !DIFile(filename: "/rustc/FAKE_PREFIX/library/core/src/ptr/mod.rs", directory: "", checksumkind: CSK_MD5, checksum: "13075ff14e9de1e7abe84070f559389f") 
         369: !162 = !DINamespace(name: "ptr", scope: !121) 
         370: !163 = !DISubroutineType(types: !164) 
         371: !164 = !{null, !138} 
         372: !165 = !{!166} 
         373: !166 = !DILocalVariable(arg: 1, scope: !160, file: !161, line: 523, type: !138) 
         374: !167 = !{!168} 
         375: !168 = !DITemplateTypeParameter(name: "T", type: !14) 
         376: !169 = !DILocation(line: 523, column: 1, scope: !160) 
         377: !170 = distinct !DISubprogram(name: "main", linkageName: "_ZN18enum_debug_niche_24main17h3169684ac0b3e294E", scope: !172, file: !171, line: 44, type: !21, scopeLine: 44, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized | DISPFlagMainSubprogram, unit: !30, templateParams: !23, retainedNodes: !173) 
         378: !171 = !DIFile(filename: "/checkout/tests/codegen/enum/enum-debug-niche-2.rs", directory: "", checksumkind: CSK_MD5, checksum: "a8dd7e92630f70c9f9bb7cb31793ab49") 
         379: !172 = !DINamespace(name: "enum_debug_niche_2", scope: null) 
         380: !173 = !{!174, !210} 
         381: !174 = !DILocalVariable(name: "x", scope: !175, file: !171, line: 45, type: !176, align: 32) 
         382: !175 = distinct !DILexicalBlock(scope: !170, file: !171, line: 45, column: 5) 
         383: !176 = !DICompositeType(tag: DW_TAG_structure_type, name: "BaseKind<enum_debug_niche_2::Declaration>", scope: !172, file: !2, size: 32, align: 32, flags: DIFlagPublic, elements: !177, templateParams: !23, identifier: "445706858b05369d80a69db83eedc0cf") 
         384: !177 = !{!178} 
         385: !178 = !DICompositeType(tag: DW_TAG_variant_part, scope: !176, file: !2, size: 32, align: 32, elements: !179, templateParams: !23, identifier: "374a1196b7c1cc46b226c9bff6b392d6", discriminator: !209) 
check:8'0                                                                                                                                                                                                            X error: no match found
         386: !179 = !{!180, !202, !207} 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
         387: !180 = !DIDerivedType(tag: DW_TAG_member, name: "Named", scope: !178, file: !2, baseType: !181, size: 32, align: 32) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         388: !181 = !DICompositeType(tag: DW_TAG_structure_type, name: "Named", scope: !176, file: !2, size: 32, align: 32, flags: DIFlagPublic, elements: !182, templateParams: !199, identifier: "203f5e6b97b20c516eb4a50ba66ff689") 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         389: !182 = !{!183} 
check:8'0     ~~~~~~~~~~~~~~~
         390: !183 = !DIDerivedType(tag: DW_TAG_member, name: "__0", scope: !181, file: !2, baseType: !184, size: 32, align: 32, flags: DIFlagPublic) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         391: !184 = !DICompositeType(tag: DW_TAG_structure_type, name: "Entity", scope: !172, file: !2, size: 32, align: 32, flags: DIFlagPublic, elements: !185, templateParams: !23, identifier: "ec80ff8feec804e99d7cf73e75282829") 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         392: !185 = !{!186} 
check:8'0     ~~~~~~~~~~~~~~~
         393: !186 = !DIDerivedType(tag: DW_TAG_member, name: "private", scope: !184, file: !2, baseType: !187, size: 32, align: 32, flags: DIFlagPrivate) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         394: !187 = !DICompositeType(tag: DW_TAG_structure_type, name: "NonZero<u32>", scope: !188, file: !2, size: 32, align: 32, flags: DIFlagPublic, elements: !190, templateParams: !197, identifier: "d4fe0445f1a0ad67d3a5abe1c89309e7") 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         395: !188 = !DINamespace(name: "nonzero", scope: !189) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         396: !189 = !DINamespace(name: "num", scope: !121) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         397: !190 = !{!191} 
check:8'0     ~~~~~~~~~~~~~~~
         398: !191 = !DIDerivedType(tag: DW_TAG_member, name: "__0", scope: !187, file: !2, baseType: !192, size: 32, align: 32, flags: DIFlagPrivate) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         399: !192 = !DICompositeType(tag: DW_TAG_structure_type, name: "NonZeroU32Inner", scope: !193, file: !2, size: 32, align: 32, flags: DIFlagPublic, elements: !194, templateParams: !23, identifier: "f023c841227c1e7561b6bcd0d1bbb950") 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         400: !193 = !DINamespace(name: "niche_types", scope: !189) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         401: !194 = !{!195} 
check:8'0     ~~~~~~~~~~~~~~~
         402: !195 = !DIDerivedType(tag: DW_TAG_member, name: "__0", scope: !192, file: !2, baseType: !196, size: 32, align: 32, flags: DIFlagPrivate) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         403: !196 = !DIBasicType(name: "u32", size: 32, encoding: DW_ATE_unsigned) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         404: !197 = !{!198} 
check:8'0     ~~~~~~~~~~~~~~~
         405: !198 = !DITemplateTypeParameter(name: "T", type: !196) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         406: !199 = !{!200} 
check:8'0     ~~~~~~~~~~~~~~~
         407: !200 = !DITemplateTypeParameter(name: "F", type: !201) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         408: !201 = !DICompositeType(tag: DW_TAG_structure_type, name: "Declaration", scope: !172, file: !2, align: 8, flags: DIFlagPublic, elements: !23, identifier: "f0fa1ff0cc8a61b0e3ffb5439145b449") 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         409: !202 = !DIDerivedType(tag: DW_TAG_member, name: "Placeholder", scope: !178, file: !2, baseType: !203, size: 32, align: 32, extraData: i32 -1) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:8'1          ?                                                                                                                                         possible intended match
         410: !203 = !DICompositeType(tag: DW_TAG_structure_type, name: "Placeholder", scope: !176, file: !2, size: 32, align: 32, flags: DIFlagPublic, elements: !204, templateParams: !199, identifier: "86d5455c6b5b7b75b7b225ebba0f6f8d") 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         411: !204 = !{!205} 
check:8'0     ~~~~~~~~~~~~~~~
         412: !205 = !DIDerivedType(tag: DW_TAG_member, name: "__0", scope: !203, file: !2, baseType: !206, align: 8, flags: DIFlagPublic) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         413: !206 = !DIBasicType(name: "!", encoding: DW_ATE_unsigned) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         414: !207 = !DIDerivedType(tag: DW_TAG_member, name: "Error", scope: !178, file: !2, baseType: !208, size: 32, align: 32, extraData: i32 0) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         415: !208 = !DICompositeType(tag: DW_TAG_structure_type, name: "Error", scope: !176, file: !2, size: 32, align: 32, flags: DIFlagPublic, elements: !23, templateParams: !199, identifier: "f7db166b9d1812807c9d7098110ee97a") 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         416: !209 = !DIDerivedType(tag: DW_TAG_member, scope: !176, file: !2, baseType: !196, size: 32, align: 32, flags: DIFlagArtificial) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         417: !210 = !DILocalVariable(name: "y", scope: !211, file: !171, line: 46, type: !80, align: 32) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         418: !211 = distinct !DILexicalBlock(scope: !175, file: !171, line: 46, column: 5) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         419: !212 = !DILocation(line: 45, column: 9, scope: !175) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         420: !213 = !DILocation(line: 46, column: 9, scope: !211) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         421: !214 = !DILocation(line: 47, column: 2, scope: !170) 
check:8'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------



```

</details>

--- a/compiler/rustc_codegen_llvm/src/debuginfo/metadata/enums/native.rs
+++ b/compiler/rustc_codegen_llvm/src/debuginfo/metadata/enums/native.rs
@@ -437,6 +437,12 @@ fn build_enum_variant_member_di_node<'ll, 'tcx>(
         .source_info
         .unwrap_or_else(|| (unknown_file_metadata(cx), UNKNOWN_LINE_NUMBER));
 
+    let discr = discr_value.opt_single_val().map(|value| {
+        let tag_base_type = tag_base_type(cx.tcx, enum_type_and_layout);
+        let size = cx.size_of(tag_base_type);
+        cx.const_uint_big(cx.type_ix(size.bits()), value)
+    });
+
     unsafe {
         llvm::LLVMRustDIBuilderCreateVariantMemberType(
             DIB(cx),
@@ -448,7 +454,7 @@ fn build_enum_variant_member_di_node<'ll, 'tcx>(
             enum_type_and_layout.size.bits(),
             enum_type_and_layout.align.abi.bits() as u32,
             Size::ZERO.bits(),
-            discr_value.opt_single_val().map(|value| cx.const_u128(value)),
+            discr,
             DIFlags::FlagZero,
             variant_member_info.variant_struct_type_di_node,
         )

--- a/tests/codegen/enum/enum-debug-niche-2.rs
+++ b/tests/codegen/enum/enum-debug-niche-2.rs
@@ -5,8 +5,8 @@
 //@ ignore-msvc
 //
 // CHECK: {{.*}}DICompositeType{{.*}}tag: DW_TAG_variant_part,{{.*}}size: 32,{{.*}}
-// CHECK: {{.*}}DIDerivedType{{.*}}tag: DW_TAG_member,{{.*}}name: "Placeholder",{{.*}}extraData: i128 4294967295{{[,)].*}}
-// CHECK: {{.*}}DIDerivedType{{.*}}tag: DW_TAG_member,{{.*}}name: "Error",{{.*}}extraData: i128 0{{[,)].*}}
+// CHECK: {{.*}}DIDerivedType{{.*}}tag: DW_TAG_member,{{.*}}name: "Placeholder",{{.*}}extraData: i32 -1{{[,)].*}}
+// CHECK: {{.*}}DIDerivedType{{.*}}tag: DW_TAG_member,{{.*}}name: "Error",{{.*}}extraData: i32 0{{[,)].*}}
 #![feature(never_type)]
 
 #[derive(Copy, Clone)]

